demos/fleet_provisioning/fleet_provisioning_with_csr/fleet_provisioning_csr_demo.c

```diff
@@ -0,0 +1,1271 @@
+                LogError( ( "Certificate buffer insufficiently large." ) );
+            }
+            else if( cborRet != CborNoError )
+            {
+                LogError( ( "Error extracting \"certificateOwnershipToken\" value from CreateCertificateFromCsr response: %s.", cbor_error_string( cborRet ) ) );
+            }
+            else
+            {
+                status = true;
+            }
+        }
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+static bool parseRegisterThingResponse( const char * response,
+                                        size_t length,
+                                        char * thingNameBuffer,
+                                        size_t * thingNameBufferLength )
+{
+    bool status = true;
+    CborError cborRet;
+    CborParser parser;
+    CborValue map;
+    CborValue value;
+
+
+    /* For details on the RegisterThing response payload format, see:
+     * https://docs.aws.amazon.com/iot/latest/developerguide/fleet-provision-api.html#register-thing-response-payload
+     */
+    cborRet = cbor_parser_init( ( const uint8_t * ) response, length, 0, &parser, &map );
+
+    if( cborRet != CborNoError )
+    {
+        LogError( ( "Error initializing parser for RegisterThing response: %s.", cbor_error_string( cborRet ) ) );
+    }
+    else if( !cbor_value_is_map( &map ) )
+    {
+        LogError( ( "RegisterThing response not a map type." ) );
+    }
+    else
+    {
+        cborRet = cbor_value_map_find_value( &map, "thingName", &value );
+
+        if( cborRet != CborNoError )
+        {
+            LogError( ( "Error searching RegisterThing response: %s.", cbor_error_string( cborRet ) ) );
+        }
+        else if( value.type == CborInvalidType )
+        {
+            LogError( ( "\"thingName\" not found in RegisterThing response." ) );
+        }
+        else if( value.type != CborTextStringType )
+        {
+            LogError( ( "\"thingName\" is an unexpected type in RegisterThing response." ) );
+        }
+        else
+        {
+            cborRet = cbor_value_copy_text_string( &value, thingNameBuffer, thingNameBufferLength, NULL );
+
+            if( cborRet == CborErrorOutOfMemory )
+            {
+                LogError( ( "Thing name buffer insufficiently large." ) );
+            }
+            else if( cborRet != CborNoError )
+            {
+                LogError( ( "Error extracting \"thingName\" value from RegisterThing response: %s.", cbor_error_string( cborRet ) ) );
+            }
+            else
+            {
+                status = true;
+            }
+        }
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+static bool waitForResponse( void )
+{
+    int i;
+    bool status = false;
+
+    responseStatus = ResponseNotReceived;
+
+    for( i = 0; i < FLEET_PROV_RESPONSE_WAIT_SECONDS; i++ )
+    {
+        ( void ) ProcessLoop( 1000 );
```

Comment:
Instead of having a wrapper around `ProcessLoop` which itself is a wrapper around `MQTT_ProcessLoop`, shouldn't the entire timeout value be provided to `ProcessLoop` so that there is only one level of indirection of making the coreMQTT API call?